### PR TITLE
feat(vss-cli): add vss vlm run

### PR DESCRIPTION
**Pull request by agmohare**
## Summary
- Adds `vss vlm run` as a synchronous command on the fixed verb grammar (`run`/`status`/`get`/`list`).
- Two media paths: Path B (`--sensor [--start-time --end-time]`) resolves a VIOS clip via `vss_core.vios`; Path A (`--media-url`) passes a pre-resolved URL directly; `--use-base64` encodes a local file inline.
- VLM endpoint is the deployment's `rt_vlm` service (discovered by `vss configure`) called via OpenAI-compatible `/v1/chat/completions`.
- Point-call design: writes one terminal memory record (status `completed`/`timeout`/`failed`) directly — no `submitted` intermediate unlike `vss summarize`.
- Intent (`--intent critic|report|qa|introspection`) is stored in `output.ext` for harness routing; `--no-persist` skips the memory write.
- Also exposes `/rtvi-vlm` in the dev-profile-lvs Ingress `cliRoutes` block (off by default, matching the dev-profile-search pattern) so `vss configure` can discover `rt_vlm` on Kubernetes.

## Test plan

- [x] `uv run --no-dev --extra cli vss vlm run --help` shows all flags
- [ ] Path A URL: `vss vlm run --prompt "..." --media-url <url>` returns `{"status":"completed","answer":"...","persisted":true}`
- [ ] Path B sensor: `vss vlm run --prompt "..." --sensor warehouse_sample` resolves VIOS clip and returns answer
- [ ] Path B windowed: `--start-time` / `--end-time` correctly clips the range
- [ ] `--no-persist` sets `"persisted": false` and skips memory write
- [ ] `--intent report` sets `"intent":"report"` in output
- [ ] `vss vlm get --job-id <id>` retrieves a persisted job
- [ ] Bad input (no source, both sources, `--start-time` without `--sensor`) → exit 2
- [ ] Timeout (`--timeout 1`) → exit 7
- [ ] Unit tests: `pytest packages/vss_cli/tests/unit_test/cli/test_vlm_group.py` — 18 tests pass
- [ ] `ruff check` + `ruff format --check` on `packages/vss_cli/src/vss_cli/vlm/` — clean

### CODE REVIEW

**Comment by agmohare** [comment]
/ok to test [1f2d4c3](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/1f2d4c38b429135620d2062851050b6a18cf6788)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33049847876)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by agmohare** [comment]
/ok to test [523fe06](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/523fe0637f5ed1187aa90f1e9e80880232b96cd3)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33094478282)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by agmohare** [comment]
/ok to test [b1cec8e](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/b1cec8e42728e0b9170754e7c4e40f4bb429706f)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33101207316)
- Internal approval evidence was evaluated by the protected compliance pipeline.

File: services/agent/packages/vss_cli/src/vss_cli/vlm/group.py [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+    )
+    end_time: str | None = Field(
+        None,
+        description="Clip window end, ISO-8601 UTC. Only valid with --sensor.",
+    )
+    media_url: str | None = Field(
+        None,
+        description="Pre-resolved HTTP/HTTPS video URL (Path A). Mutually exclusive with --sensor.",
+    )
+    intent: Literal["critic", "report", "qa", "introspection"] | None = Field(
```
Comment (zac-wang-nv) [inline services/agent/packages/vss_cli/src/vss_cli/vlm/group.py]: **P1** — The module contract says `qa` is the default and that intent is available for harness routing, but this default leaves ordinary runs with no intent in `input.intent`, the result body, or `output.ext.intent`. A caller that omits `--intent` therefore cannot be routed as QA. Please make `"qa"` the actual default (and add a persistence assertion), or remove the documented default and update downstream expectations.

File: services/agent/packages/vss_cli/src/vss_cli/vlm/group.py [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+        resolved_start: str | None = inputs.start_time
+        resolved_end: str | None = inputs.end_time
+        if inputs.sensor:
+            # Path B: fetch a clip from VIOS. Capture the effective bounds VIOS resolved,
+            # which may extend beyond the CLI inputs when only one bound was supplied.
+            try:
+                media_url, resolved_start, resolved_end = _resolve_vios_clip(
+                    deployment, inputs.sensor, inputs.start_time, inputs.end_time
+                )
+            except Exception as exc:
```
Comment (zac-wang-nv) [inline services/agent/packages/vss_cli/src/vss_cli/vlm/group.py]: **P1** — This converts every typed VIOS failure into exit 2: backend outages (`BackendUnreachableError`) no longer exit 3, timeouts no longer exit 7, and missing sensors no longer exit 5. Automation will “fix the request” instead of retrying or disambiguating. Sensor mode also bypasses the missing-service preflight because `requires` declares only `rt_vlm`, even though this path calls VST. Please conditionally require `vst` for sensor input and let the existing typed VIOS exceptions reach `guarded()`.

File: services/agent/packages/vss_cli/src/vss_cli/vlm/group.py [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+    media_url: str,
+    model: str,
+    max_tokens: int | None,
+    temperature: float | None,
+    use_base64: bool,
+    num_frames: int,
+) -> dict[str, Any]:
+    """Build an OpenAI-compatible /v1/chat/completions payload."""
+    if use_base64:
+        with open(media_url, "rb") as fh:
```
Comment (zac-wang-nv) [inline services/agent/packages/vss_cli/src/vss_cli/vlm/group.py]: **P1** — A missing, unreadable, or directory-valued `--media-url` with `--use-base64` raises an unmapped `OSError`, so the CLI exits 1 instead of the documented invalid-input exit 2. This request-building step also happens before the VLM error handling. Please validate/read the local path behind a typed `InvalidInput` boundary and add a CLI test for a nonexistent file.

File: services/agent/packages/vss_cli/src/vss_cli/vlm/group.py [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+            media_url = inputs.media_url  # type: ignore[assignment]
+
+        from vss_core.memory.adapters import utc_now_iso
+
+        from .memory_adapter import VlmAdapter
+
+        adapter = VlmAdapter()
+        created_at = utc_now_iso()
+
+        model_params: dict[str, Any] = {"model": model, "timeout": inputs.timeout}
```
Comment (zac-wang-nv) [inline services/agent/packages/vss_cli/src/vss_cli/vlm/group.py]: **P2** — `--num-frames` changes what the model sees, but it is not included in the `model_params` persisted to unified memory. Consequently, `get`/`list` records for runs with different frame budgets are indistinguishable even though their answers can differ. Please persist `num_frames` (or its wire-key spelling) and cover it in the memory-record test.

File: services/agent/packages/vss_cli/src/vss_cli/vlm/group.py [hunk 5]
```diff
@@ -0,0 +1,501 @@
...
+        return self
+
+
+class VlmOptions(BaseModel):
+    """Job and transport options. Not sent to the VLM backend."""
+
+    model_config = ConfigDict(extra="forbid")
+
+    no_persist: bool = Field(False, description="Skip writing the answer to unified memory.")
+    use_base64: bool = Field(
```
Comment (zac-wang-nv) [inline services/agent/packages/vss_cli/src/vss_cli/vlm/group.py:218]: **P1** — Please add a first-class `--file PATH` source instead of overloading `--media-url` with filesystem paths. The intended surface has three mutually exclusive sources: `--sensor` for media already in VIOS, `--media-url` for a URL that RT-VLM fetches directly, and `--file` for a local video. A `--file` run should upload/register the video with VST by default and then use the resolved clip path; `--file --use-base64` should be the explicit escape hatch that sends the bytes inline. `--media-url` should continue to mean an actual URL. Please model these three sources directly and cover both local-file branches.

Comment (zac-wang-nv) [reply services/agent/packages/vss_cli/src/vss_cli/vlm/group.py:218]: use_base64 shall work for both --file and media-url cases.

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33126111719)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by agmohare** [comment]
/ok to test [96ecd9a](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/96ecd9a0e524eafff7b92a894a780bcbada38525)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors; an OSRB-relevant required check is failing

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33141819644)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by agmohare** [comment]
/ok to test  [ece9dec](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/ece9dec7cbe5426c360fd9471735b5c3c060de6a)

**Comment by agmohare** [comment]
/ok to test [d3fc0ce](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/d3fc0cedf66dd81a897892c26bb46c858c8b6800)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33207558715)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by agmohare** [comment]
/ok to test [4c8dbb1](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/4c8dbb16e2a4e835c5d1609bc3cb84da813a96d4)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33428809799)
- Internal approval evidence was evaluated by the protected compliance pipeline.

File: deploy/helm/developer-profiles/dev-profile-lvs/templates/vss-ingress.yaml [hunk 6]
```diff
@@ -1,82 +1,45 @@
...
-{{- $_ := set $b "elasticsearch" (dict "service" (include "vss.lvs.serviceShort" (dict "root" . "short" "elasticsearch")) "port" $vi.elasticsearchPort) }}
-{{- end }}
-{{- if and $agentUmbrellaOn (include $en (dict "vals" $vaMcp "default" false)) }}
-{{- $_ := set $b "va-mcp" (dict "service" (include "vss.lvs.serviceShort" (dict "root" . "short" "vss-va-mcp")) "port" $vi.vaMcpPort) }}
-{{- end }}
-{{- if and $infraOn (include $en (dict "vals" $fp "default" true)) }}
-{{- $_ := set $b "phoenix" (dict "service" (include "vss.lvs.serviceShort" (dict "root" . "short" "phoenix")) "port" $vi.phoenixPort) }}
-{{- end }}
-{{- include "vss.ingress.assertBackends" (dict "backends" $b) }}
+{{- $cli := $vi.cliRoutes | default dict }}
```
Comment (zac-wang-nv) [inline deploy/helm/developer-profiles/dev-profile-lvs/templates/vss-ingress.yaml]: **P1** — Please keep this profile on the canonical ingress table in `deploy/helm/services/common` instead of putting the CLI mounts behind an off-by-default `cliRoutes`. With the current defaults, the rendered LVS paths omit `/rtvi-vlm`, `/elasticsearch`, and `/storage`, so `vss configure` cannot discover `rt_vlm` or unified memory, and sensor-resolved `/storage/...` media URLs fall through to the UI. `python3 deploy/helm/scripts/verify-ingress-routes.py` also fails on the four root `/v1/*` mounts and stale examples. Please rebase the RT-VLM addition on the current shared template rather than replacing it.

File: deploy/helm/developer-profiles/dev-profile-lvs/templates/vss-ingress.yaml [hunk 7]
```diff
@@ -88,28 +51,173 @@ metadata:
...
+          - path: /websocket
+            pathType: Prefix
+            backend:
+              service:
+                name: {{ include "vss.lvs.serviceShort" (dict "root" . "short" "vss-agent") }}
+                port:
+                  number: {{ $vi.vssAgentPort }}
+          # The summarization workflow uses the same public origin for LVS and
+          # RT-VLM. Keep these as exact paths: both backends own /v1 APIs.
+          - path: /v1/ready
```
Comment (zac-wang-nv) [inline deploy/helm/developer-profiles/dev-profile-lvs/templates/vss-ingress.yaml]: **P1** — These hard-coded routes ignore component enablement. Rendering with `vss-summarization.enabled=false`, `rtvi.vss-rtvi-vlm.enabled=false`, and `agent.vss-agent.enabled=false` still emits the same LVS, RT-VLM, and agent paths, all pointing at Services that are no longer created. Please preserve the `vss.ingress.enabled`-gated backend construction from `develop` so an explicit `false` removes that backend's routes.

File: services/agent/packages/vss_cli/src/vss_cli/vlm/group.py [hunk 8]
```diff
@@ -0,0 +1,521 @@
...
+            prompt=inputs.prompt,
+            sensor=inputs.sensor,
+            start_time=resolved_start,
+            end_time=resolved_end,
+            media_url=media_url if (not inputs.sensor and not _use_base64_effective) else None,
+            intent=inputs.intent,
+            model_params=model_params,
+        )
+
+        memory = self.memory(ctx) if not options.no_persist else None
```
Comment (zac-wang-nv) [inline services/agent/packages/vss_cli/src/vss_cli/vlm/group.py]: **P1** — The command-group boundary here is correct: when `--no-persist` is not specified, command groups should call `self.memory(ctx)` without inspecting `enabled` or `persist_by_default`. The memory module owns the policy established by `vss configure memory` and must decide whether persistence is active. Today that abstraction does not fully honor its configuration: `memory.build()` raises when memory is disabled and does not turn `persist_by_default=false` into a non-persisting path. Please enforce the configured policy inside the memory module so this call remains uniform across command groups and normal VLM execution proceeds with `persisted: false` when persistence is disabled.

**Comment by agmohare** [comment]
/ok to test [597bcc7](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/597bcc79165eeff20c9e5a789a4d6f46c71db6ef)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33441305356)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by agmohare** [comment]
/ok to test [ef66109](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/ef66109274de53ca91466ef3321ccbee719c04dd)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33447325143)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by agmohare** [comment]
/ok to test [fb0bf84](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/fb0bf84369502cc66865a6d4baf830e4a9a8bbfb)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33453252773)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by agmohare** [comment]
/ok to test [b2db652](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/b2db652b1d27cb0643f5ee164863eeb8ccafa976)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33459926987)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by agmohare** [comment]
/ok to test [15a770e](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/15a770e1cc242ff05ee15dc81119a84ae111a81c)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33526924028)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by agmohare** [comment]
/ok to test [0746a4e](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/0746a4e76697dbe02a8f56efd80308a4ce643f6b)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33536572464)
- Internal approval evidence was evaluated by the protected compliance pipeline.

File: services/agent/packages/vss_cli/src/vss_cli/vlm/group.py [hunk 9]
```diff
@@ -0,0 +1,717 @@
...
+        if inputs.sensor:
+            if "vst" not in (deployment.services or {}):
+                raise config_mod.ConfigError(
+                    "--sensor requires the `vst` service in the deployment. Re-run `vss configure --base-url <URL>`."
+                )
+            try:
+                media_url, resolved_start, resolved_end = _resolve_vios_clip(
+                    deployment, inputs.sensor, inputs.start_time, inputs.end_time
+                )
+            except Exception as _vios_exc:
```
Comment (zac-wang-nv) [inline services/agent/packages/vss_cli/src/vss_cli/vlm/group.py:419]: **P1** — Every post-mint outcome needs to return the framework's marker facts. This branch re-raises after creating and possibly persisting a job, so callers receive no stdout job ID or completion marker. The explicit timeout, failure, and success `Result`s below also omit `extra["marker"]`, so `_completion_marker()` defaults them to `status="completed", persisted=false`; a real timeout therefore ends with `vss_job_completed`, and a successful memory write is advertised as unpersisted. Please route all post-mint exits through an outcome helper that supplies the actual status and persistence state, and add CLI assertions for the final marker line.

File: services/agent/packages/vss_cli/src/vss_cli/vlm/group.py [hunk 10]
```diff
@@ -0,0 +1,717 @@
...
+
+def _extract_answer(completion: dict[str, Any]) -> str:
+    """Pull the text answer out of an OpenAI-style completion."""
+    try:
+        content = completion["choices"][0]["message"]["content"]
+    except (KeyError, IndexError, TypeError) as exc:
+        raise ValueError("VLM response has no choices[0].message.content") from exc
+    if not isinstance(content, str):
+        raise ValueError(f"VLM response content is not a string: {type(content).__name__}")
+    return content
```
Comment (zac-wang-nv) [inline services/agent/packages/vss_cli/src/vss_cli/vlm/group.py:131]: **P1** — Reject empty or whitespace-only content before marking the job completed. The existing `OpenAIVLMAnalyzer.analyze()` treats a blank answer as an invalid backend response, but this path persists it with `status="completed"` and exits 0. That makes media-fetch or sampling failures that surface as a silent empty answer look successful. Please require `content.strip()` and add a regression test for an empty completion.

**Comment by agmohare** [comment]
/ok to test [5dd3572](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/5dd35729c7db30be6383afb7b2063a70e8ffdd65)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors; an OSRB-relevant required check is failing

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33554833533)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by agmohare** [comment]
/ok to test [1183b63](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1883/commits/1183b6327681ac770a214989949c24132210560c)

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/33559785243)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Review by zac-wang-nv — APPROVED** [review]
LGTM
